src/Umbraco.Web/Search/UmbracoTreeSearcher.cs

```diff
@@ -234,11 +235,26 @@ private bool BuildQuery(StringBuilder sb, string query, string searchFrom, strin
 
                     foreach (var f in fields)
                     {
+                        var queryWordsReplaced = new string[querywords.Length];
+
+                        // when searching file names containing hyphens we need to replace the hyphens with spaces
```

Comment:
Perhaps fine for now but I think ideally we'd have a custom analyzer for this field?

Reply:
Agreed, I think this is fine for now, but introducing a method for custom analyzers per field is something to think about for the future

Reply:
Cool, we'll just see how this goes in the wild and do it properly later if necessary.